### DISCUSSION TITLE
how to create full slide size stretched three columns

### DISCUSSION THREAD

**Discussion by wakatana** [marp-team/marp#194] [Q&A] [answered]
Hello. I'm using marp/marpit to create slides. My goal is to create slide with three vertical columns. I have one CSS file which I use on HTML page and also in marp. In web browser the HTML page it is widen to whole page. But in PDF viewer (file generated by marp) slide has some borders from left/right top/bottom.

Files that I'm using:

**page.html**
```
<!DOCTYPE html>
<html>

<!-- page.html -->

<head>
	<!-- <link rel="stylesheet" href="css.css"> -->
	<link rel="stylesheet" href="custom-theme.css">
</head>

<body>
	<div class="container">
		<!-- <div class="toppane">Test Page</div> -->
		<div class="leftpane">
			<h1>Left column</h1>
		</div>
		<div class="middlepane">
			<h1>Middle column</h1>
		</div>
		<div class="rightpane">
			<h1>Right column</h1>
		</div>
	</div>
</body>

</html>
```

**custom-theme.css**
```
/* custom-theme.css */
/* @theme custom-theme */

@import 'default';

body, html {
	width: 100%;
	height: 100%;
	margin: 0;
  }
  
  .container {
	width: 100%;
	height: 100%;
  }
  
  .leftpane {
	  width: 33%;
	  height: 100%;
	  float: left;
	  background-color: rosybrown;
	  border-collapse: collapse;
  }
  
  .middlepane {
	  width: 34%;
	  height: 100%;
	  float: left;
	  background-color: royalblue;
	  border-collapse: collapse;
  }
  
  .rightpane {
	width: 33%;
	height: 100%;
	position: relative;
	float: right;
	background-color: yellow;
	border-collapse: collapse;
  }
  
  /* .toppane {
	width: 100%;
	height: 100px;
	border-collapse: collapse;
	background-color: #4da6ff;
  } */
 
```
 
```
---
marp: true
theme: custom-theme

---

**slide.md**
<!-- slide.md -->

<div class="container">
	<!-- <div class="toppane">Test Page</div> -->
		<div class="leftpane">
			<h1>Left column</h1>
		</div>
		<div class="middlepane">
			<h1>Middle column</h1>
		</div>
		<div class="rightpane">
			<h1>Right column</h1>
		</div>
</div>

<!-- https://stackoverflow.com/questions/36662712/how-do-i-divide-a-page-in-three-vertical-sections/36662828 -->
```

Code that I'm using to generate PDF:
```
marp --html --bespoke.progress --allow-local-files --theme-set custom-theme.css --pdf -- slide.md -o slide.pdf
```

**CSS used in HTML and displayed in browser**
![image](https://user-images.githubusercontent.com/12486963/138573026-04fac5ec-be6f-4e2f-8b47-52997de5ce00.png)

**PDF file generated by marp**
![image](https://user-images.githubusercontent.com/12486963/138573045-4cb853db-906a-438a-93c1-0ebb305ae62e.png)

How can I stretch columns to full slide size in PDF?
PS: why font has different size & color?

Thanks

**Comment by yhatt** [comment, marked as answer]
Simply remove a line `@import 'default';`. This is Marp-specific definition for [to inherit styles](https://marpit.marp.app/theme-css?id=customized-theme) from [Marp default theme](https://github.com/marp-team/marp-core/blob/main/themes/default.scss). You have inherited a lot of unique styles by this line.

### Hint

The root element for contents in common HTML is `<html>`, but _[the root element in Marp slides is `<section>` for each pages](https://marpit.marp.app/theme-css?id=html-structure)_. All of CSS declarations in Marp will apply only in rendered Marp slides. So the styles for outer elements like `<html>` and `<body>` will _be ignored in Marp rendering_.

If you want to make CSS compatible with both of HTML page and Marp slides, it's important to apply styles to both root elements. [`:root` pseudo-class](https://marpit.marp.app/theme-css?id=root-pseudo-class-selector) will match to the correct root element in both cases (match to `<html>` in common HTML, and `<section>` in Marp/Marpit).
